run.py

```diff
@@ -49,6 +49,12 @@ def run(
             dataset = Dataset(dataset_config)
             dataset.download()
             try:
+                if dataset_config["type"] == "sparse":
+                    if engine_config.get("vector_type") != "sparse":
```

Comment:
I don't really like bringing `vector_type` to config, I would rather hide in the codebase

Reply:
I agree. Introduced `supports_sparse_vectors` property in the client and removed `vector_type` from engine configs.